### PR TITLE
 profile model  created #7

### PR DESCRIPTION
**Pull request by uche-okoroafor**
### What this PR does (required):
- Created a Profile Model with Mongoose Schema

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/models/User.js [hunk 1]
```diff
@@ -19,20 +21,21 @@ const userSchema = new mongoose.Schema({
   register_date: {
```
Comment (sundayezeilo) [inline server/models/User.js:21]: Prefer camelCase in naming your variables

**Review by sundayezeilo — COMMENTED** [review]
Good work creating the user profile model. You can go ahead and start adding the endpoint.  💪🏻
Remember to always link the ticket you are working on in the PR.
Also, remember to always follow the guide in the repo to create your PR (writing a descriptive message, adding screenshot(s) of your results) so it will be easy to follow.

File: server/models/UserProfile.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    address: noReqString,
+    city: noReqString,
+    country: noReqString,
+    online: {
+      type: 'boolean',
+      required: true
+    },
+    about: reqString
+  },
+  { timestamps: true }
```
Comment (rajivtitus) [inline server/models/UserProfile.js:36]: 👍🏽

File: server/models/UserProfile.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  type: 'string',
+  required: true
+}
+
+const noReqString = {
+  type: 'string',
+  required: false
+}
+
+const userProfile = new mongoose.Schema(
```
Comment (rajivtitus) [inline server/models/UserProfile.js:17]: Don't we need an availability field too? :)

File: server/models/UserProfile.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  {
+    firstName: reqString,
+    lastName: reqString,
+    userImage: reqString,
+    nickName: noReqString,
+    dateOfBirth: { type: Date, default: Date.now },
```
Comment (rajivtitus) [inline server/models/UserProfile.js]: I don't think having the present day as a default here makes much sense :)

File: server/models/User.js (hunk 1 above)
Comment (uche-okoroafor) [reply server/models/User.js:21]: ok

File: server/models/UserProfile.js (hunk 2 above)
Comment (uche-okoroafor) [reply server/models/UserProfile.js:36]: thanks

File: server/models/UserProfile.js (hunk 3 above)
Comment (uche-okoroafor) [reply server/models/UserProfile.js:17]: i thought the "availability field" is the same thing as online Status.

File: server/models/UserProfile.js (hunk 4 above)
Comment (uche-okoroafor) [reply server/models/UserProfile.js]: ok, which year should i change the date to?

Comment (rajivtitus) [reply server/models/UserProfile.js]: I would suggest not to have any default for the `dateOfBirth` field.

File: server/models/UserProfile.js (hunk 3 above)
Comment (rajivtitus) [reply server/models/UserProfile.js:17]: The `availability` field contains information about a dog sitters availability (days of the week and the period of time  they are available for service)

On a side note, you can remove the `online` field once this is done.

Comment (uche-okoroafor) [reply server/models/UserProfile.js:17]: ok

**Pull request closed without merging**